AdyenActions/Components/3DS2/Action handlers/ThreeDS2CoreActionHandler.swift

```diff
@@ -144,24 +183,66 @@ internal class ThreeDS2CoreActionHandler: Component {
         }
 
     }
+    
+    /// Invoked to handle the error flow of a challenge handling by the 3ds2sdk.
+    /// For challenge cancelled we return the control back to the merchant immediately as an error.
+    private func didReceiveErrorOnChallenge(error: Error?,
+                                            challengeAction: ThreeDS2ChallengeAction,
+                                            completionHandler: @escaping (Result<ThreeDSResult, ThreeDS2CoreActionHandlerError>) -> Void) {
+        guard let error = error as? NSError else {
+            didFail(with: .unknown(UnknownError(errorDescription: "Both error and result are nil, this should never happen.")),
+                    completionHandler: completionHandler)
+            return
+        }
+        switch (error.domain, error.code) {
+        case (ADYRuntimeErrorDomain, Int(ADYRuntimeErrorCode.challengeCancelled.rawValue)):
+            do {
+                let cancellationResult = try ThreeDSResult(authorizationToken: challengeAction.authorisationToken,
+                                                           threeDS2SDKError: error.base64Representation())
+                let cancellationError = ThreeDS2CoreActionHandlerError.cancellationAction(cancellationResult)
+                didFail(with: cancellationError,
```

Comment:
we should make sure that the merchant will receive the same error as before, otherwise if they suddenly receive `ThreeDS2CoreActionHandlerError.cancellationAction(cancellationResult)` instead of `NSError` for cancellation then this is a breaking change, whatever the merchant used to receive as error should stay the same

Reply:
The merchant will receive a new error case for cancellation for the existing type `ThreeDS2Component.challengeCancelled`, as below. Not the `CoreActionhandler` one. 
Is there a better way? 

![image](https://github.com/Adyen/adyen-ios/assets/96450350/f1b96004-7c53-4acf-a88a-0e05de82aed7)

Reply:
We will not expose the merchant to the cancel error as it would be a breaking change.